src/classes/UTIL_CurrencyCache_TEST.cls

```diff
@@ -86,6 +90,27 @@ private class UTIL_CurrencyCache_TEST {
         System.AssertEquals(0, UTIL_CurrencyCache.cachedMapOfRatesByCurrencyCode.size(), 'The cache should be empty');
     }
 
+    /**
+     * @description Validate that the retrieveAndCacheExchangeRates method doesn't map null CurrencyData values when queried.
+     */
+    static testMethod void test_NullCurrencyDataValues() {
```

Comment:
I like this test.  Addresses the real issue.  I'd maybe change the name of this test to a more active name that includes a verb. For example `test_NullCurrencyDataIsNotCached`.  I know that naming pattern is not happening elsewhere in this class but it's the style we've been using lately in more recent code so I think new tests should adopt it.